src/main/kotlin/org/jetbrains/kotlin/jupyter/util.kt

```diff
@@ -22,11 +27,24 @@ fun <T> Logger.catchAll(msg: String = "", body: () -> T): T? = try {
 
 fun <T> T.validOrNull(predicate: (T) -> Boolean): T? = if (predicate(this)) this else null
 
+fun <T> T.asDeferred(): Deferred<T> = this.let { GlobalScope.async { it } }
+
 fun File.existsOrNull() = if (exists()) this else null
 
-fun File.readIniConfig() =
+fun <T, R> Deferred<T>.asyncLet(selector: suspend (T) -> R): Deferred<R> = this.let {
+    GlobalScope.async {
+        selector(it.await())
+    }
+}
+
+fun <T> Deferred<T>.awaitBlocking(): T = if (isCompleted) getCompleted() else runBlocking { await() }
+
+fun String.parseIniConfig() =
+        split("\n").map { it.split('=') }.filter { it.count() == 2 }.map { it[0] to it[1] }.toMap()
```

Comment:
Why isn't some INI parser used? In this implementation comments (`;`) are not handled properly. I understand that `.properties` has only one line, and it may be left as is this time.